crates/bevy_input/src/keyboard.rs

```diff
@@ -253,3 +253,213 @@ pub enum KeyCode {
     Paste,
     Cut,
 }
+
+impl KeyCode {
+    /// Converts a key into its textual char representation on a QWERTY keyboard depending on the provided shift status.
+    ///
+    /// Calls [`KeyCode::to_qwerty_char_with_shift`] if shift is down, or [`KeyCode::to_qwerty_char_without_shift`] if shift isn't down.
+    ///
+    /// Returns `None` if the key has no valid textual mapping on a QWERTY keyboard with the provided shift status.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use bevy_input::prelude::KeyCode;
+    /// #
+    /// let c1 = KeyCode::Slash.to_qwerty_char(false);
+    /// assert_eq!(c1, Some('/'));
+    ///
+    /// let c2 = KeyCode::Slash.to_qwerty_char(true);
+    /// assert_eq!(c2, Some('?'));
+    ///
+    /// let c3 = KeyCode::End.to_qwerty_char(true);
+    /// assert_eq!(c3, None);
+    /// ```
+    pub fn to_qwerty_char(self, shift_down: bool) -> Option<char> {
```

Comment:
I think calling this `qwerty` isn't quite correct. It seems like this would work equally well on an AZERTY keyboard. How about just `to_char()`?

Reply:
The chars don't match for `Key1` to `Key0` between azerty and qwerty.
For example, with azerty you have to press shift to get the number

Reply:
Arg gotcha gotcha. in that case idk if hard-coding qwerty is the right answer. I don't want to optimize bevy apis for specific regions.

Maybe we should just create a new event for [ReceivedCharacter](https://docs.rs/winit/0.23.0/winit/event/enum.WindowEvent.html#variant.ReceivedCharacter)?

Reply:
Realistically, the "correct" approach would be to have different keysets/locales which you run key codes through and which each have conversions. Localization in general is a remarkably complex problem...

Because of this, I was imagining this as a sort of "stop-gap" solution until the proper work would be done.

But looking at that winit event you linked, implementing a new input thing which hooks into that event to supply direct text would likely be the better and more cross-platform solution...